src/tasks.ts

```diff
@@ -91,3 +105,146 @@ function createDefaultTaskConfig(): object {
 
     return tasks;
 }
+
+let taskProvider: Disposable | null = null;
+
+export function activateTaskProvider(): void {
+    if (taskProvider !== null) {
+        console.log('the task provider has been activated');
+        return;
+    }
+
+    const provider: TaskProvider = {
+        provideTasks: function ()  {
+            // npm or others parse their task definitions. So they need to provide 'autoDetect' feature.
+            //  e,g, https://github.com/Microsoft/vscode/blob/de7e216e9ebcad74f918a025fc5fe7bdbe0d75b2/extensions/npm/src/main.ts
+            // However, cargo.toml does not support to define a new task like them.
+            // So we are not 'autoDetect' feature and the setting for it.
+            return getCargoTasks();
+        },
+        resolveTask(_task: Task): Task | undefined {
+            return undefined;
+        }
+    };
+
+    taskProvider = workspace.registerTaskProvider('rust', provider);
+}
+
+export function deactivateTaskProvider(): void {
+    if (taskProvider !== null) {
+        taskProvider.dispose();
+    }
+}
+
+interface CargoTaskDefinition extends TaskDefinition {
+    // FIXME: By the document, we should add the `taskDefinitions` section to our package.json and use the value of it.
+    type: 'shell';
+    taskName: string;
+    command: string;
+    args: Array<string>;
+}
+
+interface TaskConfigItem {
+    definition: CargoTaskDefinition;
+    problemMatcher?: Array<string>;
+    group?: TaskGroup;
+    presentationOptions?: TaskPresentationOptions;
+}
+
+function getCargoTasks(): Array<Task> {
+    const problemMatcher = ['$rustc'];
+
+    const presentationOptions: TaskPresentationOptions = {
+        reveal: TaskRevealKind.Always,
+        panel: TaskPanelKind.New,
+    };
+
+    const taskList: Array<TaskConfigItem> = [
+        {
+            definition: {
+                taskName: 'cargo build',
+                type: 'shell',
+                command: 'cargo',
+                args: [
+                    'build'
+                ],
+            },
+            problemMatcher,
+            group: TaskGroup.Build,
+            presentationOptions,
+        },
```

Comment:
Do we still get to be 'the' build task with this approach? I.e., do we get to call this task on ctrl + shift + b ?